src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,52 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_STORAGE_SIZE = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int size = 0;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_STORAGE_SIZE];
+        values = (V[]) new Object[MAX_STORAGE_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        int keyIndex = getKeyIndex(key);
+        if (keyIndex >= 0) {
+            values[keyIndex] = value;
+            return;
+        }
+        if (size < 10) {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        } else {
+            throw new RuntimeException("Key: '" + key + "' not found and Storage is full.");
+        }
```

Comment:
```suggestion
        if (keyIndex >= 0) {
            values[keyIndex] = value;
            return;
        } else {
             keys[size] = key;
            values[size] = value;
            size++;
        }
```

You do not need this check, and you are not asked to throw out exceptions in the problem statement